HLTrigger/HLTfilters/plugins/TriggerResultsFilter.cc

```diff
@@ -107,8 +127,13 @@ void TriggerResultsFilter::parse(const std::string& expression) {
   m_expression.reset(triggerExpression::parse(expression));
 
   // check if the expressions were parsed correctly
-  if (not m_expression)
-    edm::LogWarning("Configuration") << "Couldn't parse trigger results expression \"" << expression << "\"";
+  if (not m_expression) {
+    if (m_eventCache.shouldThrow()) {
+      throw cms::Exception("Configuration") << "Couldn't parse trigger-results expression \"" << expression << "\"";
+    } else {
```

Comment:
This is a new addition: I would say `TriggerResultsFilter` should throw an exception (not just a warning) if `throw=True` and the parsing of the expression fails.